### DISCUSSION TITLE
Built in DAP support

### DISCUSSION THREAD

**Discussion by sjcobb2022** [LazyVim/LazyVim#159] [Ideas]
I was just thinking whether there would be any interest in having built in DAP support in LazyVim.

Whilst this could be done very easily by user config, integrated DAP support is a feature that many devs consider essential for effective debugging. Using DAP in lazy would be as easy as adding nvim-dap and mason-nvim-dap.nvim so that the user could seamlessly add debugging protocols.

Additionally the implementation of nvim-dap-ui could be a awesome feature to make LazyVim more like a fully integrated IDE.

I see there was already discussion about this in #42 so I hope that we can see this awesome features implemented!

This is just a little suggestion that I thought would be cool, but I could see how others may want to leave this for the user to configure.

**Comment by appelgriebsch** [comment]
Not sure if you really need `mason-nvim-dap.nvim` as most of the functionality to download and update DAP adapters is available in `mason` itself. So you could easily refer to DAP adapters in a kind of language integration extension (see: https://github.com/appelgriebsch/Nv/blob/main/lua/plugins/extras/lang/rust.lua#L34).
